src/common/formatters.ts

```diff
@@ -93,6 +93,54 @@ export function millisecondsToHMS(valueMS: number): string {
     return parts.length ? parts.join(' ') : unknownValueString;
 }
 
+/** Outputs a value in moment.Duration in (Y M D H M S) format (ex. 1y 1M 1d 2h 3m 30s) */
+export function durationToYMWDHMS(duration: moment.Duration): string {
```

Comment:
I had to re-implement this instead of refactoring the above function. Although yearly offset is not common, monthly and daily are.